core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -732,17 +732,7 @@ class DynamicLogConfig(logManager: LogManager, server: KafkaBroker) extends Brok
   override def reconfigure(oldConfig: KafkaConfig, newConfig: KafkaConfig): Unit = {
     val originalLogConfig = logManager.currentDefaultConfig
     val originalUncleanLeaderElectionEnable = originalLogConfig.uncleanLeaderElectionEnable
-    val newBrokerDefaults = new util.HashMap[String, Object](originalLogConfig.originals)
```

Comment:
Just curious - Do we know why this logic was even added? I cant seem to understand if there was any benefit to doing this.

Reply:
See discussion here https://github.com/apache/kafka/pull/17258#discussion_r1775665743